x/bitcoin/keeper/msg_server.go

```diff
@@ -225,157 +223,159 @@ func (s msgServer) ConfirmOutpoint(c context.Context, req *types.ConfirmOutpoint
-
-		return &types.VoteConfirmOutpointResponse{
-			Status: fmt.Sprintf("outpoint %s was discarded ", req.OutPoint),
-		}, nil
-	}
-
-	s.Logger(ctx).Info(fmt.Sprintf("outpoint %s was confirmed ", req.OutPoint))
-	event = event.AppendAttributes(sdk.NewAttribute(sdk.AttributeKeyAction, types.AttributeValueConfirm))
-
-	addr, ok := s.GetAddressInfo(ctx, pendingOutPointInfo.Address)
-	if !ok {
-		return nil, fmt.Errorf("cannot confirm outpoint of unknown address")
-	}
-
-	s.SetConfirmedOutpointInfo(ctx, addr.KeyID, pendingOutPointInfo)
-
-	key, ok := s.signer.GetKey(ctx, addr.KeyID)
-	if !ok {
-		return nil, fmt.Errorf("cannot key %s", addr.KeyID)
-	}
-
-	rotationCount, ok := s.signer.GetRotationCountOfKeyID(ctx, addr.KeyID)
-	if !ok {
-		return nil, fmt.Errorf("cannot find rotation count of key %s", addr.KeyID)
-	}
-
-	currRotationCount := s.signer.GetRotationCount(ctx, exported.Bitcoin, key.Role)
-	_, nextKeyFound := s.signer.GetNextKey(ctx, exported.Bitcoin, key.Role)
-	if nextKeyFound {
-		currRotationCount++
-	}
-
-	if currRotationCount-rotationCount > s.signer.GetKeyUnbondingLockingKeyRotationCount(ctx) {
-		return &types.VoteConfirmOutpointResponse{
-			Status: fmt.Sprintf("cannot confirm outpoint of the old key %s anymore", addr.KeyID)}, nil
-	}
-
-	switch addr.Role {
-	case types.Deposit:
-		// handle cross-chain transfer
-		depositAddr := nexus.CrossChainAddress{Address: pendingOutPointInfo.Address, Chain: exported.Bitcoin}
-		amount := sdk.NewInt64Coin(types.Satoshi, int64(pendingOutPointInfo.Amount))
-		if err := s.nexus.EnqueueForTransfer(ctx, depositAddr, amount); err != nil {
-			return nil, sdkerrors.Wrap(err, "cross-chain transfer failed")
-		}
-
-		telemetry.IncrCounter(float32(pendingOutPointInfo.Amount), types.ModuleName, "total", "deposit")
-		telemetry.IncrCounter(1, types.ModuleName, "total", "deposit", "count")
-
-		recipient, ok := s.nexus.GetRecipient(ctx, depositAddr)
-		if !ok {
-			return nil, fmt.Errorf("cross-chain sender has no recipient")
-		}
-		event = event.AppendAttributes(
-			sdk.NewAttribute(types.AttributeKeyDestinationChain, recipient.Chain.Name),
-			sdk.NewAttribute(types.AttributeKeyDestinationAddress, recipient.Address),
-		)
-
-		return &types.VoteConfirmOutpointResponse{
-			Status: fmt.Sprintf("transfer of %s from {%s} successfully prepared", amount.Amount.String(), depositAddr.String()),
-		}, nil
-	case types.Consolidation:
-		unconfirmedAmount := s.BTCKeeper.GetUnconfirmedAmount(ctx, addr.KeyID)
-		s.BTCKeeper.SetUnconfirmedAmount(ctx, addr.KeyID, unconfirmedAmount-pendingOutPointInfo.Amount)
-		// the outpoint simply deposits funds into a consolidation address. Simply confirm
-		return &types.VoteConfirmOutpointResponse{
-			Status: "confirmed top up of consolidation balance"}, nil
-	default:
-		return nil, fmt.Errorf("outpoint sends funds to address with unrecognized role")
-	}
+	return nil, sdkerrors.ErrUnknownRequest
+
+	// ctx := sdk.UnwrapSDKContext(c)
```

Comment:
I assume this is supposed to be deleted?

Reply:
more like re-implemented.